### ISSUE TITLE
Option to skip loading `.env` files

### ISSUE THREAD

**Issue by einar-hjortdal** [oven-sh/bun#1564]
### What is the problem this feature would solve?

By default bun loads .env files if they exist. There should be a way to opt-out of this behavior.

### What is the feature you are proposing to solve the problem?

Add an option to bun run to opt-out of the default .env behavior

### What alternatives have you considered?

Create no .env files

This is an option, but becomes a problem when sharing code, especially open-source projects.

**Comment by Kapsonfire** [comment]
well whats the use-case? to provide an example .env file? just call it .env.example 
.env files are meant to be read, and you can define them reacting on NODE_ENV

```
bun run automatically loads environment variables from .env into the shell/task. .env files are loaded with the same priority as the rest of bun, so that means:

.env.local is first
if ($NODE_ENV === "production") .env.production else .env.development
.env
```

so just dont name the file .env or .env.local or .env.development


i dont see a usecase where you want a .env file on  project root level, but dont read it

**Comment by einar-hjortdal** [comment]
> i dont see a usecase where you want a .env file on project root level, but dont read it

One example would be debugging code that may be affected by process.env without yeeting all the .env files elsewhere and move them back.

Another is to allow providing an example without forcing the users to manually rename .env.example to .env

**Comment by Kapsonfire** [comment]
soo forcing them to run it explicitely with a '--skip-dotenv' flag is better than make them renaming the .env file?

i dont understand the debugging step... process.env is already populated by the users profile which could also affect it already.. you can overwrite .env files vars explicetly by setting them in run command afaik
`MY_ENV=123 bun script.ts`

so if you really need to debugging it with controlled ENVs, you may want to run `bun run env` and check the env's setting by users profile and/or the .env files loaded... 

more likely we want something like completely ignore all ENVs set.
`--ignore-env` <<< ignore all env set in current shell
`--ignore-dotenv` <<< ignore the .env file
`--no-env` <<< combine both

**Comment by weyert** [comment]
Yes, would be nice if it didn't automatically load dotenv files instead support `--env-file` argument similar to Node.js https://nodejs.org/dist/latest-v20.x/docs/api/cli.html#--env-fileconfig

As this appears to conflict with my `direnv` usage

**Comment by RiskyMH** [comment]
FWIW, if you pass empty string to the env file flag (`bun --env-file=""`), it doesn't load any.

**Comment by theoephraim** [comment]
@Jarred-Sumner - now that we will have a `--no-env-file` option, is there a way to programatically check the current config/flags? Tools that do their own .env loading (including [one I am building](https://github.com/dmno-dev/varlock)) may want to check to ensure that the bun flag was set correctly and throw an error (or at least warning) otherwise.
